### PR TITLE
Add middleware for checking the authorization of user

### PR DESCRIPTION
**Pull request by AndreyLykhoman**
**Description:** Add middleware for checking the authorization of user

**Youtrack:** https://youtrack.raccoongang.com/issue/DIDACTICUM-42

**Configuration instructions:**  If need of that for site you can setup site's setting (admin pages) 'DISABLE_CHECK_AUTHORIZATION' = True

**Testing instructions:**
If you are not logged in, you are redirected to the authorization page. Only Get requests are processed and the administrator authorization page is excluded.

### CODE REVIEW

File: openedx/core/djangoapps/site_configuration/middleware.py [hunk 1]
```diff
@@ -53,3 +54,26 @@ def _set_cookie_wrapper(key, value='', max_age=None, expires=None, path='/', dom
             response.set_cookie = _set_cookie_wrapper
 
         return response
+
+
+class AuthorizationCheckMiddleware(object):
+    """
+    Middleware for checking the authorization of users.
```
Comment (idegtiarov) [inline openedx/core/djangoapps/site_configuration/middleware.py]: The first line in the multiple doc strings should have the trailing empty line.

File: openedx/core/djangoapps/site_configuration/middleware.py [hunk 2]
```diff
@@ -53,3 +54,26 @@ def _set_cookie_wrapper(key, value='', max_age=None, expires=None, path='/', dom
...
+    if not authorized and  site's setting 'DISABLE_CHECK_AUTHORIZATION' is False
```
Comment (idegtiarov) [inline openedx/core/djangoapps/site_configuration/middleware.py]: New sentence should start from the capital character.

File: openedx/core/djangoapps/site_configuration/middleware.py (hunk 1 above)
Comment (AndreyLykhoman) [reply openedx/core/djangoapps/site_configuration/middleware.py]: Thank you. I did not know this, but now I will try to observe.

File: openedx/core/djangoapps/site_configuration/middleware.py (hunk 2 above)
Comment (AndreyLykhoman) [reply openedx/core/djangoapps/site_configuration/middleware.py]: I fixed that mistake. Thank's

**Pull request merged**